### DISCUSSION TITLE
Axios post request and other node modules that send api requests work locally and on github server, but not on Vercel

### DISCUSSION THREAD

**Discussion by fraserburch** [vercel/next.js#68283] [Help] [answered]
### Summary

I have tried to send a post request from my next.js app and the request works both locally and when deployed on github server when running a github action, but that same request does not work when deployed to vercel. When I say it doesn't work I don't get a 401 or a 404 or any status errors. It simply appears to not even execute the request - I try to print logs using console.log, but nothing is being logged within the request.
There are no errors nor warnings when deploying or running the app.

Here is a snippet of code I've tried (works locally, doesn't work on vercel)

`
  const _payload = {
    event: {
      message: { ResponseJson }
    },
    sourcetype: "_json"
  };

  const agent = new https.Agent({  
    rejectUnauthorized: false
  });
  const config: AxiosRequestConfig = {
    headers: _headers,
    httpsAgent: agent,
    timeout: 10000
  };
  // Send the POST request
  async function sendPostRequest() {
    try {
      const response = await axios.post('URL', _payload, config);
      // Process the response (response data type is 'any')
      console.log('Response:', response.data);
    } catch (error) {
      if (axios.isAxiosError(error)) {
        console.error('Axios error:', error);
      } else {
        console.error('Unexpected error:', error);
      }
    }
  }
  console.log('trying post request')
  sendPostRequest();
// Axios done
  console.log('done with post request')`

**Comment by nilsRagot** [comment]
Hey ! Same problem here, did you find a solution ?

**Reply by fraserburch** [reply, marked as answer]
Yea what I did was create a separate function that returned a promise. And then I awaited that function.
The way vercel handles serverless functions caused it to skip the callback, so explicitly wrapping the request in something like:
```return new Promise((resolve, reject) => {``` 
worked for me
